utils/publish-snapshot.sh

```diff
@@ -20,64 +20,97 @@ ROOT="$MYPATH/.."
 
 pushd "$ROOT"
 
-if [ -z ${CI+CI_UNSET} ]; then
-    echo "CI environment variable is unset. Exiting!"
-    exit 1
-fi
-
-if [ -z ${GITHUB_EVENT_NAME+GITHUB_EVENT_NAME_UNSET} ]; then
-    echo "GITHUB_EVENT_NAME environment variable is unset. Exiting!"
-    exit 1
-fi
-
-if [ -z ${GITHUB_REF+GITHUB_REF_UNSET} ]; then
-    echo "GITHUB_REF environment variable is unset. Exiting!"
-    exit 1
-fi
-
-if [[
-    "$GITHUB_EVENT_NAME" != "workflow_dispatch" \
-    && "$GITHUB_EVENT_NAME" != "pull_request"
-]]; then
-    exit
-fi
-
-# Check if we need to do any work
-# NOTE: changeset's --output flag has a bug where it always prefixes whatever
-# you pass to it with `cwd` (the code does `path.join(cwd, outputParam)`). So
-# we just allow it to write the file in our local dir (although I would prefer
-# to use `mktemp`).
-yarn changeset status --verbose --output changeset-status.json
-# We use jq to check if the json outpu has changesets. If not, we exit the
-# process (but not with a non-zero exit status because we don't want to cause
-# the github action to exit with a failure status).
-jq -e '.releases | length | if . > 0 then . else "No changesets found" | halt_error(1) end' \
-    < "changeset-status.json" || \
-    exit 0
-
-
-echo "Running for $GITHUB_EVENT_NAME @ $GITHUB_REF"
-
-# Example GITHUB_REF
-# refs/pull/:prNumber/merge
-if [[ "$GITHUB_REF" =~ refs/pull/([[:digit:]]+)/merge ]]; then
-    echo "Found PR #${BASH_REMATCH[1]}"
-    PR_NUMBER="PR${BASH_REMATCH[1]}"
-else
-    echo "Pull Request number not found in ref. Exiting!"
-    exit 1
-fi
-
-# publish:ci steps
-node "$ROOT/utils/pre-publish-check-ci.js"
-
-if ! git diff --stat --exit-code HEAD; then
-    echo "Git repo is dirty. This is unexpected when running in CI."
-    echo "Please review the logs leading up to this error to figure out why " \
-         "the repo was touched."
-    exit 1
-fi
-
+verify_env() {
```

Comment:
Lots of lines moved here.  I've gathered bits of functionality into functions so it's a bit less of a "massive wall of procedural functionality".  Now if you skip to line 107 you can follow the functionality more easily (I hope!).  The new lines are in `create_npmrc` (lines 88-99)!